tools/degreeworks-scraper/src/components/Scraper.ts

```diff
@@ -0,0 +1,190 @@
+import { jwtDecode } from "jwt-decode";
+import type { JwtPayload } from "jwt-decode";
+
+import type { Program } from "../types";
+
+import { AuditParser, DegreeworksClient } from ".";
+
+export class Scraper {
+  private ap!: AuditParser;
+  private dw!: DegreeworksClient;
+
+  private degrees: Map<string, string> | undefined = undefined;
+  private majorPrograms: Set<string> | undefined = undefined;
+  private minorPrograms: Set<string> | undefined = undefined;
+
+  private done = false;
+  private parsedMinorPrograms: Map<string, Program> | undefined = undefined;
+  private parsedUgradPrograms: Map<string, Program> | undefined = undefined;
+  private parsedGradPrograms: Map<string, Program> | undefined = undefined;
+  private parsedSpecializations: Map<string, Program> | undefined = undefined;
+  private degreesAwarded: Map<string, string> | undefined = undefined;
+
+  private constructor() {}
+
```

Comment:
Define the sleep utility here.
```suggestion
sleep = (ms: number = this.delay) => new Promise((r) => setTimeout(r, ms));
```